test/rules/yield-effects-spec.js

```diff
@@ -0,0 +1,219 @@
+"use strict";
+var rule = require("../../lib/rules/yield-effects");
+var RuleTester = require("eslint").RuleTester;
+
+var ruleTester = new RuleTester();
+
+function buildTest(imports, code) {
+  var result = (imports && imports + ";") || "";
+  result += "function* test() { " + code + " }";
+  return result;
+}
+
+function createTestCases(macro = "") {
+  return {
+    valid: [
+      {
+        code: buildTest(
+          `import { take } from 'typed-redux-saga${macro}'`,
+          "yield take('ACTION')"
+        ),
+      },
+      {
+        code: buildTest(
+          `import { delay } from 'typed-redux-saga${macro}'`,
+          "yield delay(1000)"
+        ),
+      },
+      {
+        code: buildTest(
+          `import { take } from 'typed-redux-saga${macro}'`,
+          "yield take('ACTION')"
+        ),
+      },
+      {
+        code: buildTest(
+          `import { take as t } from 'typed-redux-saga${macro}'`,
+          "yield t('ACTION')"
+        ),
+      },
+      {
+        code: buildTest(
+          `import { delay as d } from 'typed-redux-saga${macro}'`,
+          "yield d(1000)"
+        ),
+      },
+      {
+        // If it is an effect name but not imported from `typed-redux-saga` then its valid
+        code: buildTest(null, "take('ACTION')"),
+      },
+      {
+        // If it is an effect name but not imported from `typed-redux-saga` then its valid
+        code: buildTest(null, "delay(1000)"),
+      },
+      {
+        code: buildTest(
+          `import { take } from 'typed-redux-saga${macro}'`,
+          "notAnEffectDoesNotNeedYield()"
+        ),
+      },
+      {
+        code: `import createSagaMiddleware from 'typed-redux-saga${macro}'; const sagaMiddleware = createSagaMiddleware();`,
+      },
+      {
+        code: buildTest(
+          `import { noop } from 'typed-redux-saga${macro}'`,
+          "noop()"
+        ),
+      },
+      {
+        code: buildTest(
+          `import { call } from 'typed-redux-saga${macro}'`,
+          "const [foo, bar] = yield [call(something), call(somethingElse)]"
+        ),
+      },
+      {
+        code:
+          `import { put } from 'typed-redux-saga${macro}';\n` +
+          "expect(generator.next().value).toEqual(put({}));",
+      },
+      {
+        code: buildTest(
+          `import { call, all, delay, fetchResources } from 'typed-redux-saga${macro}'`,
+          "yield all([" +
+            "call(fetchResource, 'users')," +
+            "call(fetchResource, 'comments')," +
+            "call(delay, 1000)" +
+            "])"
+        ),
+      },
+      {
+        code:
+          `import { takeEvery } from 'typed-redux-saga${macro}';\n` +
+          "export const fooSagas = [" +
+          "takeEvery('FOO_A', fooASaga)," +
+          "takeEvery('FOO_B', fooBSaga)];",
+      },
+      {
+        code:
+          `import { call } from 'typed-redux-saga${macro}';\n` +
```

Comment:
template literals can span multiple lines:
```
const foobar = `
foo
bar`;
```

I think that would be cleaner for these code snippets.